### PR TITLE
Initial Army branch card closes #19

### LINKED ISSUE #19: Create Branch info component
Build out a branch card. This card should be able to take an image source and an items list as props.

The image will display the Branch logo.
The items will be the items in the link list below the image.

image

### PR DESCRIPTION
**Pull request by S10SEN**
Created Army Branch card component. 

Has Army image and working links.

Links were put in constants.ts then imported into component.

### CODE REVIEW

File: src/lib/constants.ts [hunk 1]
```diff
@@ -30,4 +30,11 @@ export const footerLinks = [
    {
      linkedin: 'https://www.linkedin.com',
      facebook: 'https://facebook.com',
+   }
+
+   export const armyLinks =
```
Comment (jpadwor1) [inline src/lib/constants.ts:35]: Lets use a data structure like this
export const branches = [
  {
    name: "Army",
    logo: "/usarmy.jpg",
    links: {
      main: "https://www.goarmy.com/",
      reserve: "https://www.usar.army.mil/",
      nationalGuard: "https://www.nationalguard.com/"
    }
  },
  {
    name: "Navy",
    logo: "/usnavy.jpg",
    links: {
      main: "https://www.navy.com/",
      reserve: "https://www.navyreserve.navy.mil/",
      nationalGuard: ""
    }
  },
//add remaining branch details
]

File: src/app/page.tsx [hunk 2]
```diff
@@ -45,6 +46,9 @@ export default async function Home() {
 
           <LogoutLink>Log out</LogoutLink>
         </div>
+        <div className="flex flex-row items-center gap-2">
+        <BranchCard />
```
Comment (jpadwor1) [inline src/app/page.tsx:50]: Great job, in the issue I mentioned having the ability to pass the image source as a prop and other branch specific info. If we do this we can have one <BranchCard/> and use if for Army, Navy, AF, etc. In the current configuration we have to make other cards specific to the branches.

We should map over the branches from constants.ts and render a BranchCard for each like this:

<BranchCard branch={branch}/>

File: src/components/ui/branch-card.tsx [hunk 3]
```diff
@@ -0,0 +1,37 @@
+import Image from "next/image";
+import Link from "next/link";
+import React from 'react'
+import {armyLinks} from "@/lib/constants"
+
+const BranchCard = () => {
```
Comment (jpadwor1) [inline src/components/ui/branch-card.tsx:6]: Add the branch props. They should match the type of the branches from constants.ts

interface BranchCardProps {
name: string;
logo: string;
links: {
main: string;
//etc.
}

File: src/components/ui/branch-card.tsx [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+  return (
+    <div className="flex flex-col items-center justify center bg-white text-black">
+        <div>
+            <Image
+                className=""
+                src="/usarmy.jpg"
```
Comment (jpadwor1) [inline src/components/ui/branch-card.tsx:12]: Use dynamic variable for image source.

Should be 
src={branch.logo}

File: src/components/ui/branch-card.tsx [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+            </a>
+        </div>
+        <div>
+            <a href={armyLinks.armyreserve}>
+                Army Reserve
+            </a>
+        </div>
+        <div>
+            <a href={armyLinks.armynationalguard}>
+                Army National Guard
```
Comment (jpadwor1) [inline src/components/ui/branch-card.tsx:30]: should be dynamic

File: src/components/ui/branch-card.tsx [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+            <Image
+                className=""
+                src="/usarmy.jpg"
+                alt="Army logo"
+                width={100}
+                height={100}
+              />
+        </div>
+        <div>
+            <a href={armyLinks.army}>
```
Comment (jpadwor1) [inline src/components/ui/branch-card.tsx:19]: In NextJS, try not to use <a> tags. Instead use <Link>. Import it from 'next/link' like this

import Link from 'next/link'

**Review by jpadwor1 — CHANGES_REQUESTED** [review]
Great Job

File: src/components/ui/branch-card.tsx [hunk 7]
```diff
@@ -1,37 +1,48 @@
 import Image from "next/image";
 import Link from "next/link";
 import React from 'react'
-import {armyLinks} from "@/lib/constants"
 
-const BranchCard = () => {
+ interface BranchCardProps {
```
Comment (jpadwor1) [inline src/components/ui/branch-card.tsx:5]: import Image from "next/image";
import Link from "next/link";
import React from "react";

interface BranchCardProps {
  branch: {
    name: string;
    logo: string;
    links: {
      name: string;
      url: string;
    }[];
  };
}
const BranchCard = ({ branch }: BranchCardProps) => {
  return (
    <div className="justify center flex flex-col items-center bg-white text-black">
      <div>
        <Image
          className=""
          src={branch.logo}
          alt="Logo"
          width={100}
          height={100}
        />
      </div>
      <div>
        {branch.links.map((link) => (
          <Link href={link.url}>{link.name}</Link>
        ))}
      </div>
    </div>
  );
};

export default BranchCard;

File: src/lib/constants.ts [hunk 8]
```diff
@@ -32,9 +32,59 @@ export const footerLinks = [
...
    }
 
-   export const armyLinks =
-   {
-    army: 'https://www.army.mil/',
-    armyreserve: 'https://www.usar.army.mil/',
-    armynationalguard: 'https://nationalguard.com/'
-   }
\ No newline at end of file
+   export const branches = [
```
Comment (jpadwor1) [inline src/lib/constants.ts:35]: If we put the links in an array we can map over them. We also won't need to hardcode National or Reserve into the component. 
```
export const branches = [
    {
      name: "Army",
      logo: "/usarmy.jpg",
      links: [
        { name: "Main", url: "https://www.goarmy.com/" },
        { name: "Reserve", url: "https://www.usar.army.mil/" },
        { name: "National Guard", url: "https://www.nationalguard.com/" }
      ]
    },
    {
      name: "Navy",
      logo: "/usnavy.jpg",
      links: [
        { name: "Main", url: "https://www.navy.com/" },
        { name: "Reserve", url: "https://www.navyreserve.navy.mil/" }
      ]
    },
  ]

File: src/app/page.tsx [hunk 9]
```diff
@@ -46,9 +47,8 @@ export default async function Home() {
 
           <LogoutLink>Log out</LogoutLink>
         </div>
-        <div className="flex flex-row items-center gap-2">
-        <BranchCard />
-        </div>
+        <div className="flex flex-row items-center gap-2">
```
Comment (jpadwor1) [inline src/app/page.tsx:50]: Make sure to call the component like this

to display multiple cards map over branches
  ```
  {branches.map((branch) => (
  <BranchCard branch={branch} />
  ))}
```

For just one card we have to specify which branch we want by index or find it another way.

```
<BranchCard branch={branches[0]} />
```

**Review by jpadwor1 — CHANGES_REQUESTED** [review]
I left some comments.

Change branches.links to an array of objects

Update interface BranchCardProps to accompany the new array

map over branch.links to display each Link

**Comment by S10SEN** [comment]
I think I got it this time, could use some adjustment with the images I think.

**Review by jpadwor1 — APPROVED** [review]

**Pull request merged**